code/game/machinery/computer/camera.dm

```diff
@@ -1,94 +1,119 @@
-
+	circuit = /obj/item/weapon/circuitboard/camera
+	
+	var/mapping = 0 // For the overview file (overview.dm), not used on this page
+	
+	var/list/network = list()
+	var/list/available_networks = list()
+	var/list/watchers = list() //who's using the console, associated with the camera they're on.
+
+/obj/machinery/computer/security/New() // Lists existing networks and their required access. Format: available_networks[<name>] = list(<access>)
+	generate_network_access()
 	..()
-
-/obj/machinery/computer/security/attack_ai(var/mob/user as mob)
-	return attack_hand(user)
-
-
-/obj/machinery/computer/security/check_eye(var/mob/user as mob)
-	if((get_dist(user, src) > 1 || !( user.canmove ) || user.blinded || !( current ) || !( current.status )) && (!istype(user, /mob/living/silicon)))
-		return null
-	user.reset_perspective(current)
+	
+/obj/machinery/computer/security/proc/generate_network_access()
+	available_networks["SS13"] =              list(access_hos,access_captain)
+	available_networks["Telecomms"] =         list(access_hos,access_captain)
+	available_networks["Research Outpost"] =  list(access_rd,access_hos,access_captain)
+	available_networks["Mining Outpost"] =    list(access_qm,access_hop,access_hos,access_captain)
+	available_networks["Research"] =          list(access_rd,access_hos,access_captain)
+	available_networks["Prison"] =            list(access_hos,access_captain)
+	available_networks["Labor"] =             list(access_hos,access_captain)
+	available_networks["Interrogation"] =     list(access_hos,access_captain)
+	available_networks["Atmosphere Alarms"] = list(access_ce,access_hos,access_captain)
+	available_networks["Fire Alarms"] =       list(access_ce,access_hos,access_captain)
+	available_networks["Power Alarms"] =      list(access_ce,access_hos,access_captain)
+	available_networks["Supermatter"] =       list(access_ce,access_hos,access_captain)
+	available_networks["MiniSat"] =           list(access_rd,access_hos,access_captain)
+	available_networks["Singularity"] =       list(access_ce,access_hos,access_captain)
+	available_networks["Anomaly Isolation"] = list(access_rd,access_hos,access_captain)
+	available_networks["Toxins"] =            list(access_rd,access_hos,access_captain)
+	available_networks["Telepad"] =           list(access_rd,access_hos,access_captain)
+	available_networks["TestChamber"] =       list(access_rd,access_hos,access_captain)
+	available_networks["ERT"] =               list(access_cent_specops_commander,access_cent_commander)
+	available_networks["CentComm"] =          list(access_cent_security,access_cent_commander)
+	available_networks["Thunderdome"] =       list(access_cent_thunder,access_cent_commander)
+	
+/obj/machinery/computer/security/Destroy()
+	if(watchers.len)
+		for(var/mob/M in watchers)
+			M.unset_machine() //to properly reset the view of the users if the console is deleted.
+	return ..()
+
+/obj/machinery/computer/security/check_eye(mob/user)

```

Comment:
Ahh, thank you. I looked at that, but it came up with something different. Thank you.